scrapper.py

```diff
@@ -21,21 +34,57 @@ class IncorrectNumberOfArticlesError(Exception):
     """
 
 
+class ResponseIsNotOk(Exception):
+    """
+    Response status code is not ok
+    """
+
+
 class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls, total_max_articles: int):
+        self.seed_urls = seed_urls
+        self.total_max_articles = total_max_articles
+        self.urls = []
+
+    @staticmethod
+    def _changing_the_link(link):
+        if 'http://journals.tsu.ru' not in link:
+            link = 'http://journals.tsu.ru' + link
+        return link
 
     def _extract_url(self, article_bs):
-        pass
+        urls_bs = article_bs.find('div', class_='two_thirds')
+        urls_bs = urls_bs.find_all('a')
+        urls_bs_full = [self._changing_the_link(url_bs['href']) for url_bs in urls_bs]
+        return urls_bs_full
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url_index, seed_url in enumerate(self.seed_urls):
+            response = requests.get(url=seed_url, headers=HEADERS)
+
+            if not response.ok:
+                raise ResponseIsNotOk
+
+            with open(f'{ASSETS_PATH}/seed_url_{seed_url_index}.html', 'w', encoding='utf-8') as file:
+                file.write(response.text)
+
+            with open(f'{ASSETS_PATH}/seed_url_{seed_url_index}.html', encoding='utf-8') as file:
+                response = file.read()
+
+            soup = BeautifulSoup(response, 'lxml')
+
+            articles_urls_bs = self._extract_url(soup)
+            full_list_with_urls = [url_bs for url_bs in articles_urls_bs if len(self.urls) < self.total_max_articles]
```

Comment:
I do not understand the goal ofthis comprehension. you repeat almost same iterations in the coming loop